src/main/java/com/commercetools/sync/products/utils/ProductUpdateActionUtils.java

```diff
@@ -416,8 +418,8 @@
                 } else {
                     final ProductVariant matchingOldVariant = oldProductVariantsWithMaster.get(newProductVariantKey);
                     final List<UpdateAction<Product>> updateOrAddVariant = ofNullable(matchingOldVariant)
-                        .map(oldVariant -> collectAllVariantUpdateActions(oldProduct, oldVariant, newProductVariant,
-                            attributesMetaData, syncOptions))
+                        .map(oldVariant -> collectAllVariantUpdateActions(getSameForAllUpdateActions(updateActions),
```

Comment:
I could not see what are you trying to do with reading that code, could you please consider refactoring the new collectAllVariantUpdateActions with a better name and handle the collect operation there 🤔 I assume there is a method overloading there but could not see the expected result 🤔

Reply:
I tried to keep the same naming convention since it's and overload as you mentioned and only takes one extra parameter.

Reply:
ok, sounds gound.